converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/ImagingStudy/ImagingStudyPropertySynchronizer.cs

```diff
@@ -30,17 +48,47 @@ public void Synchronize(FhirTransactionContext context, ImagingStudy imagingStud
                 return;
             }
 
-            ImagingStudyPipelineHelper.SetDateTimeOffSet(context);
+            SynchronizePropertiesAsync(imagingStudy, context, false, AddStartedElement, cancellationToken);
+            SynchronizePropertiesAsync(imagingStudy, context, false, AddImagingStudyEndpoint, cancellationToken);
+            SynchronizePropertiesAsync(imagingStudy, context, false, AddModality, cancellationToken);
+            SynchronizePropertiesAsync(imagingStudy, context, false, AddNote, cancellationToken);
+            SynchronizePropertiesAsync(imagingStudy, context, false, AddAccessionNumber, cancellationToken);
+        }
 
-            AddStartedElement(imagingStudy, dataset, context.UtcDateTimeOffset);
-            AddImagingStudyEndpoint(imagingStudy, context);
-            AddModality(imagingStudy, dataset);
-            AddNote(imagingStudy, dataset);
-            AddAccessionNumber(imagingStudy, dataset);
+        private void SynchronizePropertiesAsync(ImagingStudy imagingStudy, FhirTransactionContext context, bool required, Action<ImagingStudy, FhirTransactionContext> synchronizeAction, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                synchronizeAction(imagingStudy, context);
+            }
```

Comment:
This function SynchronizePropertiesAsync feels repeated many times can it be abstracted

Reply:
Thanks, I made it a generic and move it to a helper class